routing/payment_lifecycle.go

```diff
@@ -12,6 +12,20 @@ import (
 	"github.com/lightningnetwork/lnd/routing/route"
 )
 
+// errNoRoute is returned when all routes from the payment session have been
+// attempted.
+type errNoRoute struct {
+	// lastError is the last attempt error, if at least one attempt has been
+	// made.
+	lastError error
```

Comment:
Hm, can we make this of type `ForwardingError` now?

Reply:
Yes that is possible. It was already possible before. Fixed.